packages/maskbook/src/extension/background-script/EthereumServices/providers/MetaMask.ts

```diff
@@ -63,6 +64,11 @@ export async function requestAccounts() {
     return accounts
 }
 
+export async function checkIfMetaMaskLocked() {
+    const accounts: string[] = await provider!.request({ method: 'eth_requestAccounts', params: [] })
```

Comment:
I think that also applies if there is really no accounts in MetaMask, or user doesn't grant MetaMask wallet access to Maskbook